src/index.ts

```diff
@@ -153,7 +153,11 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
   options.cacheControl = Object.assign({}, defaultCacheControl, evalCacheOpts)
 
   // override shouldEdgeCache if options say to bypassCache
-  if (options.cacheControl.bypassCache || options.cacheControl.edgeTTL === null) {
+  if (
+    options.cacheControl.bypassCache ||
+    options.cacheControl.edgeTTL === null ||
+    request.method == 'HEAD'
```

Comment:
It will result in all `HEAD` request become cacheable, which is not the desired behavior.

Reply:
Not sure I follow - this is literally the opposite.

Reply:
> Not sure I follow - this is literally the opposite.

Sorry, my fault, typo. I meant to say `not-cacheable`.

Reply:
[RFC 7231](https://tools.ietf.org/html/rfc7231#section-4.3.2) affirms that `HEAD` is cacheable so I agree with you in that regard but Workers KV does not support it. As illustrated in https://github.com/cloudflare/kv-asset-handler/issues/140 all `HEAD` requests to Workers KV that request edge caching result in an exception. So, Workers KV can start supporting `HEAD`, or this this library can correctly handle the fact that it doesn't 🤷‍♂️.